crates/test-programs/src/bin/preview2_tcp_connect.rs

```diff
@@ -76,30 +76,16 @@ fn test_tcp_connect_dual_stack(net: &Network) {
     v4_listener.blocking_listen().unwrap();
 
     let v4_listener_addr = v4_listener.local_address().unwrap();
-    let v6_listener_addr =
-        IpSocketAddress::new(IpAddress::IPV4_MAPPED_LOOPBACK, v4_listener_addr.port());
 
     let v6_client = TcpSocket::new(IpAddressFamily::Ipv6).unwrap();
 
     // Tests:
 
-    // Even on platforms that don't support dualstack sockets,
-    // setting ipv6_only to true (disabling dualstack mode) should work.
-    v6_client.set_ipv6_only(true).unwrap();
-
-    // Connecting to an IPv4-mapped-IPv6 address on an ipv6-only socket should fail:
+    // Connecting to an IPv4 address on an IPv6 socket should fail:
     assert!(matches!(
-        v6_client.blocking_connect(net, v6_listener_addr),
+        v6_client.blocking_connect(net, v4_listener_addr),
```

Comment:
This should attempt to connect to `v6_listener_addr` which was a IPv4-mapped-IPv6 address. The way you rewrote it, this call doesn't actually test the change.

Reply:
Got it - thanks!